protocol-units/mempool/util/src/lib.rs

```diff
@@ -1,172 +1,224 @@
 use serde::{Deserialize, Serialize};
 
-use movement_types::{Block, Transaction, Id};
+use movement_types::{Block, Id, Transaction};
 use std::cmp::Ordering;
+use thiserror::Error;
+
+pub type BoxedStoreError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+#[derive(Error, Debug)]
+pub enum MempoolTransactionOperationsError {
```

Comment:
Let's make naming less verbose and erase `Operations` everywhere except the trait names.
```suggestion
pub enum MempoolTransactionError {
```

Reply:
I agree with this. `-Operations` should only be a suffix for traits and only where there's not another obvious name.

Reply:
The trait names are indeed a bit verbose too, so maybe we can come up with terser names all around.
I don't see why errors need to be prefixed by the trait name, though. If anything, I'd like to see package names used instead of verbose prefixes, as this is more in Rust style. So you could have `mempool::TransactionOperations` (re-exported from `mempool::transaction`) and `mempool::transaction::Error`.

Reply:
I will leave it for now, and after we merge #71, we can settle on a naming convention for these public APIs.